src/test/compile-fail/empty-struct-unit-expr.rs

```diff
@@ -24,10 +24,10 @@ enum E {
 fn main() {
     let e2 = Empty2(); //~ ERROR expected function, found `Empty2`
     let e4 = E::Empty4();
-    //~^ ERROR `E::Empty4` is being called, but it is not a function
+    //~^ ERROR expected a function, but `E::Empty4` is `E`
```

Comment:
Even if `E::Empty4` is declared without arguments it still can be called if `E` implements `Fn` or friends. Something like
```
`E::Empty4` cannot be called, because `E` doesn't implement `Fn`
```
will tell the reason precisely.